AudioCompression/src/audioCompression/algorithm/AlgorithmStep.java

```diff
@@ -22,18 +22,24 @@
 	 * 
 	 * @param input
 	 *            the input data for the forward operation
+	 * @param name
+	 *      	  the name of the file to be written, this gets passed through 
+	 *      	  each stage mostly for writing out to disk
 	 * @return the forward processed data
 	 */
-	public OutputType forward(InputType input);
+	public OutputType forward(InputType input, String name);
```

Comment:
Not so sure about this one... This is for tracking a filename, right? Could we maybe require a filename in the constructor for the IO related steps instead?